common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResult.java

```diff
@@ -22,7 +22,117 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
 
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResponse;
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_CODE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_FINAL_URL;
 
 public class AzureActiveDirectoryAuthorizationResult extends AuthorizationResult {
+
+    private static final String TAG = AzureActiveDirectoryAuthorizationResult.class.getSimpleName();
+
+    public AzureActiveDirectoryAuthorizationResult(int resultCode, final Intent data) {
```

Comment:
Please restrict the functionality of constructors to setting fields (and optionally logging) _only_.

I'll save the rant and link you to [page 3 of Wolter, Ruffer, and Hevery's 'Writing Testable Code'](https://drive.google.com/file/d/1tXHR-0vzeyNcgzlzFjT1VroRkQqa56xt/view?usp=sharing). Chances are, this logic belongs elsewhere.

Consider using a factory or builder to create these objects, preferably using constructors which accept arguments specific to the way(s) this class needs to get initialized. Avoid calling `new` inside constructors, as this creates dependencies which can't be practically mocked. For instance, this constructor creates instances of `AuthorizationErrorResponse` -- this inter-class dependency isn't clear from this constructor's API and doesn't allow for testing to substitute alternative implementations of `AuthorizationErrorResponse`. Ideally, the relationship between these classes should be expressed as an interface like `IAuthorizationErrorResponse` in the event that `AuthorizationErrorResponse` needs to become `MicrosoftAuthorizationErrorResponse` or `AmazonAuthorizationErrorResponse` in the future